phases/10-llms-from-scratch/05-scaling-distributed/code/main.py

```
@@ -0,0 +1,339 @@
+        loss, grad = model_fn(shard)
+        gpu_losses.append(loss)
+        gpu_gradients.append(grad)
+
+    avg_loss = np.mean(gpu_losses)
+    avg_gradient = np.mean(gpu_gradients, axis=0)
+
+    return avg_loss, avg_gradient
+
+
+def simulate_tensor_parallelism(input_data, weight_matrix, num_gpus):
+    d_in, d_out = weight_matrix.shape
```

Ruff (0.15.7)

[warning] Unpacked variable `d_in` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)